evm/src/cross_table_lookup.rs

```diff
@@ -293,14 +384,17 @@ pub(crate) fn eval_cross_table_lookup_checks<F, FE, P, C, S, const D: usize, con
             next_z,
             challenges,
             columns,
-            filter_columns,
+            filter_column,
         } = lookup_vars;
-        let combine = |v: &[P]| -> P { challenges.combine(columns.iter().map(|&i| &v[i])) };
+        let combine = |v: &[P]| -> P {
+            let evals = columns.iter().map(|c| c.eval(v)).collect::<Vec<_>>();
```

Comment:
Can we skip collecting here?

Reply:
I'm not sure. `challenge.combine` takes a `IntoIterator<Item = &'a P>` (same as `reduce_with_powers`). So if I try changing this to 
```rust
let evals = columns.iter().map(|c| &c.eval(v));
challenges.combine(evals)
```
I get a "cannot return reference to temporary value" error. One solution would be to change `reduce_with_powers` to take a `IntoIterator<Item = impl Borrow<P>>`. I've tried this and the downside is that we then have to add explicit generics to every call of `reduce_with_powers`. If you think it's worth it in terms of performance, that could be a solution.

Reply:
Oh I see. I thought `Borrow` params could typically be inferred? But in any case feel free to merge as is, I can play with it later.